### ISSUE TITLE
Keep `default-detekt-config.yml` synced is a mess

### ISSUE THREAD

**Issue by BraisGabin** [detekt/detekt#7746]
## Expected Behavior
I don't need to execute a command and commit it's output when working with detekt.

## Current Behavior
When I modify things related with rule configurations or add a new rule I need to execute `generateDocumentation` task and commit the changes.

## Context
#7745 is a case where this strange behavior makes the contributions more difficult for the maintainers and for the person opening the PR too.

Even I get this type of errors. I know how to fix them fast but it would be nice to avoid them in the first place.

## Proposal
Don't commit `default-detekt-config.yml` anymore. Configure the `gradle` tasks correctly so the file is generated when needed.

I see a pro and a con with this.

- Pro: we get users that find the `default-detekt-config.yml` on our repo and uses it instead of using the task `generateConfig`. This creates issues because the file `default-detekt-config.yml` is in sync with `main` instead of the version that the user is using. Users had open issues related with this. If we remove the file this problem can't happen
- Con: The configuration changes are easier to review (on PRs) from the `yml` file than from the code that it generates it. But if this is a huge problem I imagine that we could find another way to solve it.

**Comment by cortinico** [comment]
> I don't need to execute a command and commit it's output when working with detekt.

Isn't this the same behavior as a lot of other tools (to name one: `binary-compatibility-validator` and the .api files).

If we move to auto-generating `default-detekt-config.yml` without committing we'll lose a lot of niceties coming from having that file in the git history. For example now you can easily look through the history of all the changes in the detekt repo that resulted in a configuration change.

**Comment by cortinico** [comment]
> I don't need to execute a command and commit it's output when working with detekt.

Isn't this the same behavior as a lot of other tools (to name one: `binary-compatibility-validator` and the .api files).

If we move to auto-generating `default-detekt-config.yml` without committing we'll lose a lot of niceties coming from having that file in the git history. For example now you can easily look through the history of all the changes in the detekt repo that resulted in a configuration change.

**Comment by BraisGabin** [comment]
You need to do the same but I think that it's different. No one wants to change the `.api` nor a `baseline.xml`. But you need to have in the repo the current state of the API or to the "known issues". Otherwise when you change the code there is no way to know how the api was before or which ones are the fresh code smells.

In the case of the `default-detekt-config.yml` we don't care that much about how it was before. We don't want to compare how it was vs how it is in the current commit.

> For example now you can easily look through the history of all the changes in the detekt repo that resulted in a configuration change.

You are right. We will lose some things. But I think that they are minor use cases. For example, in the past we also commit the documentation (the task that generates the `default-detekt-config.yml` is still called `generateDocumentation` for this fact). When we stopped we "lost" exactly the same but with the changes on the documentation.

TL;DR: Yes, we will lose some things but I think that the change will help us more than hurt. The idea behind this is to make the project easier to contribute. If we don't want this solution we could use `danger` to give advice when the workflow `verify-generated-config-file` fails.

**Comment by BraisGabin** [comment]
We should take a decision here. I don't have a hard opinion on this, we can just close the issue but I would like to know what other maintainers think. Would this make detekt easier to maintain or more difficult? There are clearly some tradeoffs here as cortinico pointed out.
